test/src/test/java/jenkins/model/ParameterizedJobMixInTest.java

```diff
@@ -62,4 +62,23 @@ public void doBuildWithParameters_shouldFailWhenInvokingDisabledProject() throws
         final JenkinsRule.WebClient webClient = j.createWebClient();
         webClient.assertFails(project.getUrl() + "buildWithParameters", HttpServletResponse.SC_CONFLICT);
     }
+
+    @Test
+    @Issue("JENKINS-48770")
+    public void doBuildQuietPeriodInSeconds() throws Exception {
+        final int projectQuietPeriodInSeconds = 10;
+        final int projectQuietPeriodInMillis = projectQuietPeriodInSeconds * 1000;
+        final int timeToExecuteProject = 5000; // Time to finish the execution of the job
+        final FreeStyleProject project = j.createFreeStyleProject();
+        project.setQuietPeriod(projectQuietPeriodInSeconds);
+
+        final JenkinsRule.WebClient webClient = j.createWebClient();
+        webClient.goTo(project.getUrl() + "build", "");
+
+        Assert.assertEquals(0, project.getBuilds().size());
+        Thread.sleep(projectQuietPeriodInMillis / 2);
+        Assert.assertEquals(0, project.getBuilds().size());
+        Thread.sleep(projectQuietPeriodInMillis / 2 + timeToExecuteProject);
```

Comment:
This test seems really long (execution duration of ~15 seconds) for what it accomplishes.  I wonder whether it couldn't be sped up.

Reply:
Queue items tell you how long they're in the queue for. So just set the quiet period to 50 (seconds), trigger a build, ask the `Queue.WaitingItem#timestamp` whether it'll wait for at least another 45000 millis, and then just abort it?

(If it's not a `WaitingItem` something else will have gone wrong. Typically `LeftItem`, meaning the build has already started.)